manager/src/main/java/com/redhat/service/bridge/manager/ProcessorServiceImpl.java

```diff
@@ -77,22 +82,32 @@ public Processor createProcessor(String bridgeId, String customerId, ProcessorRe
             throw new AlreadyExistingItemException("Processor with name '" + processorRequest.getName() + "' already exists for bridge with id '" + bridgeId + "' for customer '" + customerId + "'");
         }
 
+        Processor newProcessor = new Processor();
+
         Set<BaseFilter> requestedFilters = processorRequest.getFilters();
+
         String requestedTransformationTemplate = processorRequest.getTransformationTemplate();
         BaseAction requestedAction = processorRequest.getAction();
-        BaseAction resolvedAction = actionProviderFactory.resolve(requestedAction, bridge.getId(), customerId);
+        ActionProvider actionProvider = actionProviderFactory.getActionProvider(requestedAction.getType());
+
+        BaseAction resolvedAction = actionProviderFactory.resolve(requestedAction,
+                bridge.getId(),
+                customerId,
+                newProcessor.getId());
+
+        newProcessor.setName(processorRequest.getName());
+        newProcessor.setSubmittedAt(ZonedDateTime.now());
+        newProcessor.setStatus(BridgeStatus.REQUESTED);
+        newProcessor.setBridge(bridge);
 
         ProcessorDefinition definition = new ProcessorDefinition(requestedFilters, requestedTransformationTemplate, requestedAction, resolvedAction);
+        newProcessor.setDefinition(definitionToJsonNode(definition));
 
-        Processor p = new Processor();
-        p.setName(processorRequest.getName());
-        p.setDefinition(definitionToJsonNode(definition));
-        p.setSubmittedAt(ZonedDateTime.now());
-        p.setStatus(BridgeStatus.REQUESTED);
-        p.setBridge(bridge);
+        processorDAO.persist(newProcessor);
 
-        processorDAO.persist(p);
-        return p;
+        connectorService.createConnectorIfNeeded(resolvedAction, newProcessor, actionProvider);
```

Comment:
@lucamolteni cc @kostola  I wonder here if we should be using the outbox pattern to create/monitor the deployment of Connectors (and other resources) see: https://debezium.io/blog/2019/02/19/reliable-microservices-data-exchange-with-the-outbox-pattern/ In particular, please see the `Sending Events to the Outbox` part of ^

So in that model we would do something like:

1. Persist the Processor here with the resolved Action
2. Fire an Event that a Processor has been requested
3. Something listens for the Event and persists the need to create a Connector to fulfil the request
4. Another process creates the Connector and then calls back the ProcessorService when the Connector is ready
5. At that point we expose details to the Shard that it can start to provision the Executor

It's out of scope for this PR, but I want to raise the pattern/approach before we get too much further into the implementation so that we can have a conversation about adopting it.

Reply:
I agree with it @robpblake